src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -82,38 +108,138 @@ def _run_pyscf(self) -> Tuple[gto.Mole, scf.hf.SCF]:
             )
         return molecule, mean_field_object
 
+    def get_occupied_and_active_indicies_with_FNO(
+        self,
+    ) -> Tuple[openfermion.MolecularData, List[int], List[int]]:
+        """
+        Reduce the virtual space with the frozen natural orbital (FNO)
+        approach and get occupied and active orbital indicies needed for
+        computing the fermionic Hamiltonian.
+
+        Returns:
+            molecular data: A PyscfMolecularData object.
+            occupied_indices: A list of molecular orbitals not in the active space.
+            active_indicies: A list of molecular orbitals to include in the active space. # noqa:E501
+        """
+        molecular_data = self._get_molecular_data()
+        mean_field_object = molecular_data._pyscf_data["scf"]
+
+        if molecular_data.multiplicity != 1:
+            raise ValueError("RO-MP2 is not available.")
+
+        n_frozen_core_orbitals = 0
+
+        if self.freeze_core:
+            mp2 = self._get_frozen_core_orbitals(molecular_data)
+            n_frozen_core_orbitals = mp2.frozen
+
+        elif self.occupied_indices and not self.freeze_core:
```

Comment:
Yes, you understood it correctly. 
I improved it based on your comments. :) When both freeze_core and occupied_indices are set, we're raising the error. Is this something you'd expect?

Reply:
Looks good! Thank you for fixing it!

Reply:
I've been thinking we might want to use the dataclass `__post_init` method to validate the arguments since it seems like there might be quite a few edge cases that aren't valid or don't make sense. (We don't need to do that in this PR though.)